### ISSUE TITLE
How to get file path inside plugin

### ISSUE THREAD

**Issue by grammka** [thejameskyle/babel-handbook#90]
Hi! Is there any way to get file path (the file, which code receiving in visitor) in plugin?

**Comment by hzoo** [comment]
There's a file object - in the visitor `Identifier(path, state) () { ... }` you can do `path.hub.file` or `state.file`

`state.file.log.filename` gives you the filename

Check astexplorer for more info http://astexplorer.net/#/Q5Kl4AV0hl

**Comment by grammka** [comment]
@hzoo I know about this, but all time I get `unknown` =(

**Comment by jridgewell** [comment]
Right, on astexplorer, you will get `unknown`. On your computer, it'll be the correct filename.

**Comment by grammka** [comment]
@jridgewell I know it.. I tried this **on my computer**. Try yourself

Here is what I did **on my computer**:

```javascript
Program: {
  enter: function(path, state) {
    console.log(path.hub.file.log.filename);
    console.log(state.file.log.filename);
  }
},
Identifier: function(path, state) {
  console.log(path.hub.file.log.filename);
  console.log(state.file.log.filename);
}
```

output:

```
unknown
unknown
unknown
...
```
